### PR TITLE
Add `collaborator` role to organization

### PR DESCRIPTION
**Pull request by mustard-mh**
## Description

PR changes are mostly because of code gen (New Year code contribution gift 🤣)

Left to right: owner, member, collaborator
SCR-20240111-raqk

TODO:
- [ ] Document PR

## Related Issue(s)

Fixes EXP-1081

## How to test

- Preview env https://hw-collaborator.preview.gitpod-dev.com/workspaces
- Feature Flag [[link](https://app.configcat.com/08da1258-64fb-4a8e-8a1e-51de773884f6/08da1258-6541-4fc7-8b61-c8b47f82f3a0/08da1258-6512-4ec0-80a3-3f6aa301f853?settingId=257865)]

**UX**

- Create an org, and invite your another account as collaborator, or join my org as a `collaborator` via https://hw-collaborator.preview.gitpod-dev.com/orgs/join?inviteId=4428fc65-1a33-4d4e-b9b3-bd8a2aa37817
- Create another org (in that org you're `owner`)
- Can't access members, usage, project of `collaborator` role org
- Can access mentioned above of `owner` role org

> To test all new member as collaborator with your org (link shared above already did it), you may need to add your organizationId to [Feature Flag](https://app.configcat.com/08da1258-64fb-4a8e-8a1e-51de773884f6/08da1258-6541-4fc7-8b61-c8b47f82f3a0/08da1258-6512-4ec0-80a3-3f6aa301f853?settingId=257865). That's because it's preview env, if it's Dedicated, this step should work automatically via cell-id

Try update db to test with `member` role

**APIs**
- Open this PR with gitpod
- Exec `cd test`
- Follow what's written in `collaborator_test.go` https://github.com/gitpod-io/gitpod/blob/4050a8cb5a8ee755a7fd5e3d69156455ce9689e0/test/tests/smoke-test/collaborator_test.go#L31-L51

#### Preview status

gitpod:summary

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [ ] /werft with-preview
- [ ] /werft with-large-vm
- [x] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [x] /werft preemptible
      Saves cost. Untick this only if you're really sure you need a non-preemtible machine.
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
- [ ] with-monitoring
</details>

/hold

### CODE REVIEW

File: components/dashboard/src/teams/OrgSettingsPage.tsx [hunk 1]
```diff
@@ -43,7 +43,7 @@ export function OrgSettingsPage({ children }: OrgSettingsPageProps) {
     const subtitle = "Manage your organization's settings.";
 
     // Render as much of the page as we can in a loading state to avoid content shift
-    if (org.isLoading || orgBillingMode.isLoading) {
+    if (org.isLoading) {
```
Comment (mustard-mh) [inline components/dashboard/src/teams/OrgSettingsPage.tsx:46]: With `orgBillingMode.isLoading`, collaborator `/billing` router (TeamUsageBasedBilling.tsx) will keep re-rendering forever.

Don't know what side-affect this change will cause. But if we consider collaborator should not visit /billing, then we can revert code here.

File: components/dashboard/src/teams/JoinTeam.tsx [hunk 2]
```diff
@@ -27,13 +29,17 @@ export default function JoinTeamPage() {
                 const result = await organizationClient.joinOrganization({ invitationId: inviteId });
                 orgInvalidator();
 
-                history.push(`/members?org=${result.organizationId}`);
+                if (!isDataOps) {
```
Comment (svenefftinge) [inline components/dashboard/src/teams/JoinTeam.tsx:32]: This is not dataops dependent but whether the user is a collaborator. I don't think redirecting new joiners to the mebers list is a good idea anyway. so let's redirect everyone to the main page.

File: components/server/src/authorization/authorizer.ts [hunk 3]
```diff
@@ -299,11 +299,27 @@ export class Authorizer {
...
+                // TODO: owner has all permission member's permission should define in schema
+                // we should remove member but since old code was not, keep it to avoid breaking
+                set(rel.organization(orgID).member.user(userID)),
+                remove(rel.organization(orgID).collaborator.user(userID)),
+            );
+        } else if (role === "member") {
+            updates.push(
+                remove(rel.organization(orgID).owner.user(userID)),
+                set(rel.organization(orgID).member.user(userID)),
+                remove(rel.organization(orgID).collaborator.user(userID)),
```
Comment (svenefftinge) [inline components/server/src/authorization/authorizer.ts:315]: I believe this should also set the `collaborator` role. It seems like we model the roles as inclusive at least the removeOrgRole method below suggests that? I.e. if would removing the `member` role be a downgrade to `collaborator`?

**Review by svenefftinge — APPROVED** [review]
LATM (Looks awesome to me). 🙏

**Pull request merged**